### PR TITLE
Kits: Fix - After save a global the new style is missing.

### PR DESCRIPTION
**Pull request by matipojo**

### CODE REVIEW

File: assets/dev/js/editor/data/globals/component.js [hunk 1]
```diff
@@ -4,11 +4,16 @@ import ColorsComponent from './colors/component';
...
+const ControlsCSSParser = require( 'elementor-editor-utils/controls-css-parser' );
+
 export default class Component extends ComponentBase {
 	__construct( args = {} ) {
 		super.__construct( args );
 
 		elementorCommon.elements.$window.on( 'elementor:loaded', this.onElementorLoaded.bind( this ) );
+
+		elementor.on( 'document:loaded', () => this.updateTempStylesheet() );
+		elementor.on( 'preview:loaded', () => this.updateTempStylesheet() );
```
Comment (kobizz) [inline assets/dev/js/editor/data/globals/component.js:16]: Do you really need to do it twice, or you may set a flag to indicate when both document and preview were loaded

**Review by kobizz — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/data/globals/component.js (hunk 1 above)
Comment (matipojo) [reply assets/dev/js/editor/data/globals/component.js:16]: It's for two different situations. `document:loaded` is used in order to remove\render on switch between to kit and back.
the `preview:loaded` is used to re-render on switch to canvas and back.

File: assets/dev/js/editor/data/globals/component.js [hunk 2]
```diff
@@ -30,4 +35,55 @@ export default class Component extends ComponentBase {
...
+	 */
+	updateTempStylesheet() {
+		if ( ! this.controlsCSS ) {
+			this.controlsCSS = new ControlsCSSParser( {
+				id: 'e-kit-temp-variables',
+				settingsModel: new elementorModules.editor.elements.models.BaseSettings( {}, {} ),
+			} );
+		}
+
+		// The kit document has it's own CSS.
```
[the remark this answers is not in the document]
Comment (matipojo) [reply assets/dev/js/editor/data/globals/component.js]: Done

File: assets/dev/js/editor/data/globals/component.js [hunk 3]
```diff
@@ -30,4 +35,55 @@ export default class Component extends ComponentBase {
 		// Add globals to cache before render.
 		$e.data.get( 'globals/index' ).then( () => elementor.trigger( 'globals:loaded' ) );
 	}
+
+	/**
+	 * In case there is a new global color/typography convert current globals to CSS variables.
```
[the remark this answers is not in the document]
Comment (matipojo) [reply assets/dev/js/editor/data/globals/component.js]: Done

File: core/kits/assets/js/globals/global-select-behavior.js [hunk 4]
```diff
@@ -279,8 +281,6 @@ export default class GlobalControlSelect extends Marionette.Behavior {
 
 		this.createNewGlobal( globalMeta )
 			.then( ( result ) => {
-				elementor.kitManager.refreshKitCssFiles();
```
[the remark this answers is not in the document]
Comment (matipojo) [reply core/kits/assets/js/globals/global-select-behavior.js:282]: Done

File: assets/dev/js/editor/data/globals/component.js (hunk 1 above)
Comment (kobizz) [reply assets/dev/js/editor/data/globals/component.js]: What will happen in the first loading? It seems like both of the events will fire and the CSS will be updated twice.

**Review by kobizz — APPROVED** [review]

**Pull request merged**